documentation/book/proc-instrumenting-producers-consumers-opentracing.adoc

```diff
@@ -0,0 +1,164 @@
+// Module included in the following assemblies:
+//
+// assembly-instrumenting-kafka-clients-tracers.adoc
+
+[id='proc-instrumenting-producers-consumers-for-opentracing-{context}']
+= Instrumenting Kafka Producers and Consumers for tracing
+
+Use a Decorator pattern or Interceptors to instrument your Java Producer and Consumer application code for distributed tracing.
+
+.Procedure
+
+Perform these steps in the application code of each Kafka Producer and Consumer application.
+
+. Add the Maven dependency for OpenTracing to the producer or consumer's pom.xml file.
++
+[source,xml,subs="+quotes,attributes+"]
+----
+<dependency>
+    <groupId>io.opentracing.contrib</groupId>
+    <artifactId>opentracing-kafka-client</artifactId>
+    <version>{0.1.5}</version>
+</dependency>
+----
+
+. If you prefer to use a Decorator pattern, instrument your client application code as shown in the following example. If you prefer to use Interceptors, go to the next step.
```

Comment:
I wonder if it would be clearer that you do one of the other if it was a single steps with bullets showing teh two options?

Reply:
Agreed.